# **1. Resumen**

Bellabeat es una empresa de bienestar fundada en el 2014 por Urška Sršen y Sandro Mur. Los productos de la empresa están pensados principalmente para las mujeres, enfocados en la recopilación de datos biométricos para poder ofrecer recomendaciones a quienes utilicen sus productos y que estas logren tener un impacto positivo en su salud y bienestar.

La gama de productos comercializados, tanto dispositivos inteligentes, como su app, está enfocada principalmente en llevar un registro de los ciclos menstruales, patrones de sueño, niveles de estrés y actividad diaria.

Actualmente (2024), Bellabeat tiene aproximadamente 10 millones de usuarios a nivel global, con más de 1 millón de dispositivos sincronizados .

El propósito del proyecto es el descubrimiento de nuevas oportunidades de crecimiento para Bellabeat, logrado a través de un análisis en el uso de dispositivos similares fabricados por la competencia.

# **2. Definición del problema**

## 2.1 Tarea empresarial

Objetivo: Describir el uso de otros dispositivos inteligentes. Encontrar patrones en el uso de otros dispositivos inteligentes enfocados en la salud y bienestar para poder implementar acciones dentro de Bellabeat.
Este análisis estará basado en datos recopilados en relojes inteligentes de FitBit.


## 2.2 Partes interesadas/*stakeholders*

Urška Sršen: cofundadora y CPO (Directora creativa de producto) de Bellabeat.
Sandro Mur: cofundador y CEO (Director ejecutivo) de Bellabeat.
Bellabeat Marketing Analytics Team / Equipo de análisis de datos de marketing.

# **3. Preparación de datos**

## 3.1 Conjunto de datos

Los datos analizados provienen del registro de actividad diaria a través de diversos dispositivos inteligentes de Fitbit.
Estos datos son de dominio público (licencia CC0 1.0), obtenidos de Kaggle a través del usuario Möbius . Al ser de dominio público, los autores del conjunto de datos otorgan su permiso para que los datos recopilados también puedan ser modificados, copiados, procesados y distribuidos libremente.


## 3.2 Origen de los datos

El conjunto de datos original  fue recopilado por medio de una encuesta distribuida por Amazon Mechanical Turk, en donde 30 usuarios prestaron su consentimiento para el envío de datos personales de seguimiento de actividad física en minutos, ritmo cardiaco, patrones de sueño y estimado de calorías quemadas a lo largo del día. Estos datos brindan la oportunidad de conocer algunos hábitos que los usuarios podrían tener, como conductas o preferencias individuales. Las diferencias entre la manera en que se presentan algunos datos dentro de los archivos pueden deberse al uso de distintos dispositivos y configuraciones definidas por los usuarios.

## 3.3 Organización de los datos y verificación

Los datos recopilados consisten en información registrada de manera diaria por 30 usuarios, disponible en 18 archivos con extensión .csv, en el periodo comprendido entre marzo-12-2016 y mayo-12-2016 (dos meses).

La estructura de los datos incluye un registro por punto temporal, es decir, los archivos incluyen varios registros para un solo usuario (celdas repetidas), en donde cada registro corresponde a un día o minuto en específico. Cada usuario tiene un id único, lo cual permite la identificación individual para facilitar el procesamiento de datos y su análisis.

Las fechas para el inicio del registro no coinciden en todos los usuarios, así como el número de registros obtenidos para cada uno. Posiblemente esto se deba a la fecha en la que cada usuario aceptó participar en la recopilación de datos y que no todos los usuarios mantuvieron su registro diario o su participación en el programa.

Para un mejor entendimiento inicial de los datos, primero fueron estudiados a través de Microsoft Excel, conociendo su estructura, número total de registros y una familiarización con las variables, en donde se generaron tablas dinámicas para obtener un resumen rápido de la información disponible en cada archivo.

| ARCHIVOS (_.csv_) | DESCRIPCIÓN |
|:----------|:----------|
| dailyActivity_merged    | Registros de actividad física diaria proveniente de 33 usuarios.<br>Información agrupada para pasos, tiempo en actividad y distancia<br>recorrida. No todos los usuarios tienen el mismo número de días<br>registrados (min: 4 días, max : 31 días).   |
| dailyCalories_merged    | Registro diario de las calorías quemadas por 33 usuarios.   |
| dailyIntensities_merged    | Registro de intensidades de actividad (tiempo y distancia) diaria de<br> 33 usuarios.   |
| dailySteps_merged    | Conteo de pasos diarios de 33 usuarios.   |
| heartrate_seconds_merged    | Ritmo cardiaco de 7 usuarios registrado durante varias veces por<br> cada minuto del día.   |
| hourlyCalories_merged    | Estimado de calorías consumidas cada hora por 33 usuarios.   |
| hourlyIntensities_merged    | Total y promedio de niveles de intensidad en la actividad física<br> durante cada hora en 33 usuarios.   |
| hourlySteps_merged    | Conteo total de pasos en cada hora registrados por 33 usuarios.   |
| minuteCaloriesNarrow_merged    | Estimado de calorías consumidas cada minuto por 33 usuarios<br> (un minuto en cada fila).   |
| minuteCaloriesWide_merged    | Estimado de calorías consumidas desglosado por cada minuto de<br> cada hora por 33 usuarios (un minuto en cada columna).   |
| minutelntensitiesNarrow_merged    | Nivel de intensidad en la actividad física por minuto de 33 usuarios<br> (un minuto en cada fila).   |
| minuteIntensitiesWide_merged    | Nivel de intensidad en la actividad física, desglosado por cada <br>minuto en cada hora de 33 usuarios (un minuto en cada columna).   |
| minuteMETsNarrow_merged    | Equivalente metabólico (tasa de energía utilizada en reposo)<br> registrado por minuto en 33 usuarios.   |
| minuteSleep_merged    | Registro por minuto del sueño en 24 usuarios. No se indica la<br> unidad de medida.   |
| minuteStepsNarrow_merged    | Conteo de pasos por minuto en 33 usuarios (un minuto en cada<br> fila).   |
| minuteStepsWide_merged    | Conteo de pasos por minuto en cada hora en 33 usuarios (un minuto<br> en cada columna).   |
| sleepDay_merged    | Registro diario en 24 usuarios del tiempo total permaneciendo<br> acostado, dormido y el número de registros por día.   |
| weightLogInfo_merged    | 67 registros de peso en kg y libras durante 30 días. Cálculo de BMI.<br>Registros obtenidos de 8 usuarios, los cuales 5 lo hicieron manualmente.   |

## 3.4 Veracidad e integridad de los datos

El conjunto de datos proporcionado para llevar a cabo la tarea empresarial presenta diversas características que pueden representar un problema al momento de ser analizados.
En primer lugar, el tamaño de muestra, siendo pequeña, con un registro inicial de 35 usuarios, entre los cuales, no todos tienen el mismo número de días registrados (algunos con menos de diez registros), puede representar un factor de sesgo, siendo posible un análisis estadístico erróneo; así como el desconocimiento de cómo fueron recopilados éstos y de quienes provienen. Es decir, no se conoce cierta información personal y otros detalles socioeconómicos y culturales de las personas a partir de las cuáles fue obtenida la información, como el género, edad, ubicación y ocupación, entre otras características. Es por esto por lo que los resultados del análisis de los datos proporcionados realmente no podrían ser específicos para un target en el desarrollo de campañas de marketing o para conocer el comportamiento de un grupo específico de la población.

En segundo lugar, el conjunto de datos proporcionado cuenta con información recopilada durante dos meses del primer semestre del 2016, por lo que puede llegar a ser obsoleta. La última actualización del conjunto de datos y su descripción en Kaggle se realizó en marzo del 2024, sin embargo, se mantienen los registros del 2016.

Los avances tecnológicos y las nuevas necesidades han facilitado el desarrollo de nuevas funcionalidades en dispositivos inteligentes que aún no existían en ese entonces, por lo que no podrían ser estudiadas todas las maneras en las que los usuarios utilizan sus dispositivos actualmente.

Sin embargo, es posible conocer, de manera muy general, cuáles podrían ser las funcionalidades básicas más utilizadas por ciertos usuarios de dispositivos inteligentes enfocados en la salud y algunos de sus hábitos. Debido a estas razones, el presente análisis será únicamente descriptivo y enfocado en los datos suministrados.

# **4. Procesamiento de datos (ETL)**

Debido al tamaño de los archivos, se decidió procesarlos y analizarlos a través de Python, gracias a su capacidad computacional para la ejecución de las tareas necesarias como limpieza, análisis y visualización.


##4.1 Herramienta de trabajo y librerías utilizadas


Primero se procedió con la importación de las librerías que serán utilizadas : Re, para el procesamiento de cadenas de texto (ReGex), Numpy y Pandas para el procesamiento numérico, de limpieza y manejo de la información, y Matplotlib y Seaborn para la visualización de datos.

In [40]:
import numpy as np
import pandas as pd
import re
import matplotlib.pyplot as ply
import seaborn as sns

## 4.2 Importación de archivos

A continuación, debido al contenido de los archivos, se eligieron únicamente aquellos que se consideraron con información relevante para el enfoque en este proyecto.

Estos fueron:  

*   _dailyActivity_merged.csv_
*   _hourlySteps_merged.csv_
*   _sleepDay_merged.csv_



Los archivos "_weightLogInfo_merged.csv_" no fue considerado para el análisis debido a que únicamente cuenta con registros de 8 usuarios. Si bien el archivo "_sleepDay_merged.csv_" únicamente contiene registros de 24 usuarios y no tiene el mismo tamaño de muestra que los otros archivos (al menos 30 usuarios), se utilizó como práctica para escribir código.

La información completa de los archivos "_dailyActivity_merged.csv_" y "_hourlySteps_merged.csv_" se encuentra en dos archivos separados: el primero comprende los registros entre el 12 de marzo de 2016 y el 11 de abril de 2016, y el segundo los registros entre el 12 de abril del 2016 y el 12 de mayo del 2016.
Por lo tanto, se procedió a unificarlos en uno solo.

In [41]:
# registros entre 12-marzo-2016 y 11-abril-2016
daily_act1 = pd.read_csv('/content/Fitbit analysis/dailyActivity_merged1.csv')

# registros entre 12-abril-2016 y 12-mayo-2016
daily_act2 = pd.read_csv('/content/Fitbit analysis/dailyActivity_merged.csv')

dailyact = pd.concat([daily_act1, daily_act2], keys= [1, 2])

In [42]:
# registros entre 12-marzo-2016 y 11-abril-2016
steps1 = pd.read_csv('/content/Fitbit analysis/hourlySteps_merged1.csv')

# registros entre 12-abril-2016 y 12-mayo-2016
steps2 = pd.read_csv('/content/Fitbit analysis/hourlySteps_merged.csv')

hourlysteps = pd.concat([steps1, steps2], keys= [1, 2])

In [43]:
# importando los registros de sueño
sleepday = pd.read_csv('/content/Fitbit analysis/sleepDay_merged.csv')


## 4.3 Primer acercamiento a los datos

Para conocer la estructura, variables y tipos de datos en cada archivo se utilizaron funciones y métodos como .head(), .columns, .shape e .info().
Como resumen, los dataframes generados incluyen la siguiente información:


1.   **dailyact**: registro de actividad diaria en 35 usuarios durante hasta dos meses.
2.   **hourlysteps**: conteo de pasos de 35 usuarios durante cada hora por hasta dos meses.
3.   **sleepday**: registro de sueño diario en 24 usuarios durante hasta dos meses.







In [44]:
print(sleepday.shape)
sleepday.columns

(413, 5)


Index(['Id', 'SleepDay', 'TotalSleepRecords', 'TotalMinutesAsleep',
       'TotalTimeInBed'],
      dtype='object')

In [45]:
sleepday.info()
sleepday.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  413 non-null    int64 
 1   SleepDay            413 non-null    object
 2   TotalSleepRecords   413 non-null    int64 
 3   TotalMinutesAsleep  413 non-null    int64 
 4   TotalTimeInBed      413 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.3+ KB


Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

In [46]:
dailyact.info()
dailyact.head()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 1397 entries, (1, 0) to (2, 939)
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        1397 non-null   int64  
 1   ActivityDate              1397 non-null   object 
 2   TotalSteps                1397 non-null   int64  
 3   TotalDistance             1397 non-null   float64
 4   TrackerDistance           1397 non-null   float64
 5   LoggedActivitiesDistance  1397 non-null   float64
 6   VeryActiveDistance        1397 non-null   float64
 7   ModeratelyActiveDistance  1397 non-null   float64
 8   LightActiveDistance       1397 non-null   float64
 9   SedentaryActiveDistance   1397 non-null   float64
 10  VeryActiveMinutes         1397 non-null   int64  
 11  FairlyActiveMinutes       1397 non-null   int64  
 12  LightlyActiveMinutes      1397 non-null   int64  
 13  SedentaryMinutes          1397 non-null   int64  
 14 

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
1 0  1503960366    3/25/2016       11004           7.11             7.11   
  1  1503960366    3/26/2016       17609          11.55            11.55   
  2  1503960366    3/27/2016       12736           8.53             8.53   
  3  1503960366    3/28/2016       13231           8.93             8.93   
  4  1503960366    3/29/2016       12041           7.85             7.85   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
1 0                       0.0                2.57                      0.46   
  1                       0.0                6.92                      0.73   
  2                       0.0                4.66                      0.16   
  3                       0.0                3.19                      0.79   
  4                       0.0                2.16                      1.09   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
1 0                 4.07                      0.0                 33   
  1                 3.91                      0.0                 89   
  2                 3.71                      0.0                 56   
  3                 4.95                      0.0                 39   
  4                 4.61                      0.0                 28   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
1 0                   12                   205               804      1819  
  1                   17                   274               588      2154  
  2                    5                   268               605      1944  
  3                   20                   224              1080      1932  
  4                   28                   243               763      1886

In [47]:
hourlysteps.info()
hourlysteps.head()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 46183 entries, (1, 0) to (2, 22098)
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Id            46183 non-null  int64 
 1   ActivityHour  46183 non-null  object
 2   StepTotal     46183 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 1.9+ MB


Id           ActivityHour  StepTotal
1 0  1503960366  3/12/2016 12:00:00 AM          0
  1  1503960366   3/12/2016 1:00:00 AM          0
  2  1503960366   3/12/2016 2:00:00 AM          0
  3  1503960366   3/12/2016 3:00:00 AM          0
  4  1503960366   3/12/2016 4:00:00 AM          0

## 4.4 Limpieza  y transformación de datos

Resultó necesaria la modificación de ciertos datos en los archivos importados, con el fin de eliminar inconsistencias entre la información disponible en cada uno y errores en el formato de algunos datos.

### 4.4.1 Eliminación de datos duplicados

Primero, se decidió verificar el numero de usuarios con registros en cada archivo, y posteriormente, en caso de contar con registros duplicados, estos fueron eliminados.

In [48]:
print('Users in dailyact:', len(dailyact['Id'].unique()))
print('Users in hourlysteps:', len(hourlysteps['Id'].unique()))
print('Users in sleepday:', len(sleepday['Id'].unique()))

Users in dailyact: 35
Users in hourlysteps: 35
Users in sleepday: 24


In [49]:
# encontrando el número de filas duplicadas en dailyact
dailyact_duplicate = pd.DataFrame(dailyact.duplicated())
dailyact_duplicate.iloc[:].value_counts()


False    1397
Name: count, dtype: int64

In [50]:
# encontrando el número de filas duplicadas en hourlysteps
hourlysteps_duplicate = pd.DataFrame(hourlysteps.duplicated())
hourlysteps_duplicate.iloc[:].value_counts()

False    46008
True       175
Name: count, dtype: int64

En el dataframe de _hourlysteps_ se encontraron 175 duplicados, por lo cual se identificaron las filas repetidas dentro del conjunto de datos y verificar tal resultado.

In [51]:
# se agregó una etiqueta a las filas duplicadas para su selección
hourlysteps['Duplicate'] = hourlysteps.duplicated()
hourlysteps[hourlysteps['Duplicate']==True]

Id           ActivityHour  StepTotal  Duplicate
2 717    1624580081  4/12/2016 12:00:00 AM         31       True
  718    1624580081   4/12/2016 1:00:00 AM          0       True
  719    1624580081   4/12/2016 2:00:00 AM          0       True
  720    1624580081   4/12/2016 3:00:00 AM          7       True
  721    1624580081   4/12/2016 4:00:00 AM          0       True
...             ...                    ...        ...        ...
  21368  8877689391   4/12/2016 4:00:00 AM          0       True
  21369  8877689391   4/12/2016 5:00:00 AM          0       True
  21370  8877689391   4/12/2016 6:00:00 AM        209       True
  21371  8877689391   4/12/2016 7:00:00 AM        964       True
  21372  8877689391   4/12/2016 8:00:00 AM        834       True

[175 rows x 4 columns]

En "_hourlysteps_" las 175 filas que se encontraron como duplicados, fueron generadas por un traslape de datos entre el primer y segundo archivo al contener registros de la madrugada del 12 de abril, por lo que se eliminaron del dataframe original. Al eliminar estos duplicados permanecieron 46, 008 registros.

In [52]:
# se eliminó la columna 'Duplicates' debido a que las etiquetas True/False eliminaban el estatus de duplicado
hourlysteps.drop('Duplicate', inplace=True, axis =1)
#eliminando las filas duplicadas
hourlysteps.drop_duplicates(inplace=True)
# y verificando el número de datos restantes
hourlysteps.info()

<class 'pandas.core.frame.DataFrame'>
MultiIndex: 46008 entries, (1, 0) to (2, 22098)
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Id            46008 non-null  int64 
 1   ActivityHour  46008 non-null  object
 2   StepTotal     46008 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 1.9+ MB


Para el dataframe "_sleep_" se llevó a cabo el mismo procedimiento para eliminar las filas duplicadas.

In [53]:
# encontrando el número de filas duplicadas en sleep
sleep_duplicate = pd.DataFrame(sleepday.duplicated())
sleep_duplicate.iloc[:].value_counts()

# se agregó una etiqueta a las filas duplicadas (3 filas) para su selección y verificación
sleepday['Duplicate'] = sleepday.duplicated()
sleepday[sleepday['Duplicate']==True]

# y después fueron eliminados
sleepday.drop('Duplicate', inplace=True, axis =1)
sleepday.drop_duplicates(inplace=True)
sleepday.info()


<class 'pandas.core.frame.DataFrame'>
Index: 410 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  410 non-null    int64 
 1   SleepDay            410 non-null    object
 2   TotalSleepRecords   410 non-null    int64 
 3   TotalMinutesAsleep  410 non-null    int64 
 4   TotalTimeInBed      410 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 19.2+ KB


In [54]:
# reseting the index and deleting any missing values there might be
dailyact = dailyact.droplevel(0)

In [55]:
dailyact.dropna(inplace=True)
dailyact.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1397 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        1397 non-null   int64  
 1   ActivityDate              1397 non-null   object 
 2   TotalSteps                1397 non-null   int64  
 3   TotalDistance             1397 non-null   float64
 4   TrackerDistance           1397 non-null   float64
 5   LoggedActivitiesDistance  1397 non-null   float64
 6   VeryActiveDistance        1397 non-null   float64
 7   ModeratelyActiveDistance  1397 non-null   float64
 8   LightActiveDistance       1397 non-null   float64
 9   SedentaryActiveDistance   1397 non-null   float64
 10  VeryActiveMinutes         1397 non-null   int64  
 11  FairlyActiveMinutes       1397 non-null   int64  
 12  LightlyActiveMinutes      1397 non-null   int64  
 13  SedentaryMinutes          1397 non-null   int64  
 14  Calories      

In [56]:
hourlysteps = hourlysteps.droplevel(0)
hourlysteps.dropna(inplace=True)

In [57]:
hourlysteps.head()

Id           ActivityHour  StepTotal
0  1503960366  3/12/2016 12:00:00 AM          0
1  1503960366   3/12/2016 1:00:00 AM          0
2  1503960366   3/12/2016 2:00:00 AM          0
3  1503960366   3/12/2016 3:00:00 AM          0
4  1503960366   3/12/2016 4:00:00 AM          0

### 4.4.2 Filtrado de muestra

El dataframe _dailyact_ contiene la información de 35 usuarios. Sin embargo, no existe el mismo número de registros para cada usuario. para evitar la influencia de valores extremos y un sesgo durante el análisis, se realizó un conteo del número de registros existente para cada uno de los usuarios, para posteriormente ser filtrados aquellos usuarios con menos de 30 días de registro,

In [58]:
records_by_user = dailyact.groupby('Id')['ActivityDate'].count().sort_values().to_frame('Number of records')
records_by_user.reset_index(inplace=True)
records_by_user

Id  Number of records
0   2891001357                  8
1   6391747486                  9
2   3372868164                 30
3   8253242879                 31
4   2347167796                 33
5   6775888955                 35
6   4057192912                 36
7   7007744171                 38
8   6117666160                 38
9   8583815059                 39
10  6290855005                 39
11  4388161847                 39
12  1644430081                 40
13  5577150313                 41
14  8792009665                 41
15  8053475328                 42
16  3977333714                 42
17  7086361926                 43
18  8378563200                 43
19  4558609924                 43
20  5553957443                 43
21  2873212765                 43
22  2320127002                 43
23  2026352035                 43
24  2022484408                 43
25  1927972279                 43
26  1844505072                 43
27  4319703577                 43
28  8877689391                 43
29  6962181067                 45
30  4445114986                 46
31  4702921684                 46
32  1624580081                 50
33  1503960366                 50
34  4020332650                 63

Debido a que los usuarios "_2891001357_" y "_6391747486_" únicamente cuentan con 8 y 9 registros, respectivamente, fueron excluidos del análisis, con lo cual se mantuvo una muestra de 33 usuarios.

In [69]:
# lista con usuarios conservados
users_to_keep = list(records_by_user['Id'][2:])
# aplicar filtro al dataframe
dailyact_filter = dailyact.loc[dailyact['Id'].isin(users_to_keep)]
dailyact_filter.reset_index(inplace=True)
print('There are', len(dailyact_filter['Id'].unique()), 'users in the "dailyact" dataframe')

There are 33 users in the "dailyact" dataframe


### 4.4.3 Estandarización de columnas

Para facilitar la unión de tablas para el análisis, se modificaron los nombres de algunas columnas entre las tablas utilizadas. Además, se eliminaron las mayúsculas en los nombres de estas (uppercase-lowercase).

In [71]:
print('1:',dailyact_filter.columns)
print('2:',hourlysteps.columns)
print('3:',sleepday.columns)

1: Index(['index', 'Id', 'ActivityDate', 'TotalSteps', 'TotalDistance',
       'TrackerDistance', 'LoggedActivitiesDistance', 'VeryActiveDistance',
       'ModeratelyActiveDistance', 'LightActiveDistance',
       'SedentaryActiveDistance', 'VeryActiveMinutes', 'FairlyActiveMinutes',
       'LightlyActiveMinutes', 'SedentaryMinutes', 'Calories'],
      dtype='object')
2: Index(['Id', 'ActivityHour', 'StepTotal'], dtype='object')
3: Index(['Id', 'SleepDay', 'TotalSleepRecords', 'TotalMinutesAsleep',
       'TotalTimeInBed'],
      dtype='object')
